Resources/public/js/recommendationsupport.js

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+
+(function (global, doc) {
+    var eZ = global.eZ = global.eZ || {};
+
+    /**
+     * Contains logic needed to fetch and display YooChoose recommendations
+     *
+     * @class
+     */
+    eZ.RecommendationSupport = function () {};
+
+    /**
+     * Default settings
+     *
+     * @type {Array}
+     */
+    eZ.RecommendationSupport.prototype.config = {
+        msgEmpty: 'No recommendations found',
+        msgError: 'An error occurred while loading the recommendations',
+        msgNotSupported: 'Cannot display recommendations, this browser is not supported',
+        restUrl: 'recommendations/fetch',
+        limit: 5
+    };
+
+    /**
+     * Overwrite default settings
+     *
+     * @method setOptions
+     * @param config {Array} user settings
+     */
+    eZ.RecommendationSupport.prototype.setOptions = function (config) {
+        for (var setting in config)
+            this.config[setting] = config[setting];
+    };
+
+    /**
+     * Return available XMLHttpRequest object (depending on browser)
+     *
+     * @method getXMLHttpRequest
+     * @returns {Object} XMLHttpRequest
+     */
+    eZ.RecommendationSupport.prototype.getXMLHttpRequest = function () {
+        var xmlHttp;
+
+        if (window.XMLHttpRequest)
```

Comment:
since you are passing the *global* scope in the anonymous function, it should be:
```js
if ( global.XMLHttpRequest ) {
```